src/main/java/org/folio/entlinks/service/authority/AuthorityService.java

```diff
@@ -0,0 +1,116 @@
+package org.folio.entlinks.service.authority;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.apache.commons.lang3.StringUtils;
+import org.folio.entlinks.domain.entity.Authority;
+import org.folio.entlinks.domain.entity.AuthoritySourceFile;
+import org.folio.entlinks.domain.repository.AuthoritySourceFileRepository;
+import org.folio.entlinks.domain.repository.AuthorityStorageRepository;
+import org.folio.entlinks.exception.AuthorityNotFoundException;
+import org.folio.entlinks.exception.AuthoritySourceFileNotFoundException;
+import org.folio.entlinks.exception.RequestBodyValidationException;
+import org.folio.spring.data.OffsetRequest;
+import org.folio.tenant.domain.dto.Parameter;
+import org.springframework.data.domain.Page;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+@Log4j2
+public class AuthorityService {
+
+  private final AuthorityStorageRepository repository;
+  private final AuthoritySourceFileRepository sourceFileRepository;
+
+  public Page<Authority> getAll(Integer offset, Integer limit, String cql) {
+    log.info("getAll:: Attempts to find all Authority by [offset: {}, limit: {}, cql: {}]", offset, limit,
+        cql);
+
+    if (StringUtils.isBlank(cql)) {
+      return repository.findAll(new OffsetRequest(offset, limit));
+    }
+
+    return repository.findByCql(cql, new OffsetRequest(offset, limit));
+  }
+
+  public Authority getById(UUID id) {
+    log.debug("getById:: Loading Authority by ID [id: {}]", id);
+
+    return repository.findById(id).orElseThrow(() -> new AuthorityNotFoundException(id));
+  }
+
+  public Authority create(Authority entity) {
+    log.debug("create:: Attempting to create Authority [entity: {}]", entity);
+
+    UUID id = entity.getId();
+    if (id != null && repository.existsById(id)) {
+      throw new RequestBodyValidationException("AuthoritySourceFile already exists",
+          List.of(new Parameter("id").value(String.valueOf(id))));
+    }
+
+    if (entity.getAuthoritySourceFile() != null && entity.getAuthoritySourceFile().getId() != null) {
+      var sourceFile = sourceFileRepository.findById(entity.getAuthoritySourceFile().getId())
+          .orElseThrow(() -> new AuthoritySourceFileNotFoundException(entity.getAuthoritySourceFile().getId()));
+      sourceFile.addAuthorityStorage(entity);
+    }
```

Comment:
I think this could be skipped. Having just an id should be enough for hibernate

Reply:
Database constraint we could handle on APIErrorHandler

Reply:
Same in update

Reply:
Done, both in create and update, added tests to check these cases